frappe/public/js/frappe/form/controls/date.js

```diff
@@ -9,37 +9,41 @@ frappe.ui.form.ControlDate = class ControlDate extends frappe.ui.form.ControlDat
 		this.set_datepicker();
 		this.set_t_for_today();
 	}
+	/**
+	 * @param {string} value
+	 * @returns {Date}
+	 */
+	value_to_date(value) {
+		return frappe.datetime.str_to_obj(value);
+	}
 	set_formatted_input(value) {
-		if (value === "Today") {
-			value = this.get_now_date();
+		if (!value) {
+			super.set_formatted_input("");
+			this.datepicker?.clear();
+			return;
 		}
 
 		super.set_formatted_input(value);
 		if (this.timepicker_only) return;
-		if (!this.datepicker) return;
-		if (!value) {
-			this.datepicker.clear();
+		if (!this.datepicker) {
+			// There is no datepicker, so we cannot set the selected date
 			return;
 		}
 
-		let should_refresh = this.last_value && this.last_value !== value;
+		let should_refresh = false;
 
-		if (!should_refresh) {
-			if (this.datepicker.selectedDates.length > 0) {
-				// if date is selected but different from value, refresh
-				const selected_date = moment(this.datepicker.selectedDates[0]).format(
-					this.date_format
-				);
-
-				should_refresh = selected_date !== value;
-			} else {
-				// if datepicker has no selected date, refresh
-				should_refresh = true;
+		if (this.datepicker.selectedDates.length > 0) {
+			const selected_date = this.datepicker.selectedDates[0];
+			const deltaTime = selected_date - this.value_to_date(value);
+			if (Math.abs(deltaTime) > 0) {
```

Comment:
Is this necessary? Shouldn’t `deltaTime != 0` already do the trick? Do I miss something?